probe/overlay/weave.go

```diff
@@ -83,6 +106,59 @@ func (w Weave) update() (weaveStatus, error) {
 	return result, json.NewDecoder(resp.Body).Decode(&result)
 }
 
+type psEntry struct {
+	containerIDPrefix string
+	macAddress        string
+	ips               []string
+}
+
+func (w Weave) ps() ([]psEntry, error) {
+	var result []psEntry
+	cmd := ExecCommand("weave", "--local", "ps")
+	out, err := cmd.StdoutPipe()
+	if err != nil {
+		return result, err
+	}
+	if err := cmd.Start(); err != nil {
+		return result, err
+	}
+	defer func() {
+		if err := cmd.Wait(); err != nil {
+			log.Printf("Weave tagger, cmd failed: %v", err)
+		}
+	}()
+	scanner := bufio.NewScanner(out)
+	for scanner.Scan() {
+		line := scanner.Text()
+		groups := weavePsMatch.FindStringSubmatch(line)
+		if len(groups) == 0 {
+			continue
+		}
+		containerIDPrefix, macAddress, ips := groups[1], groups[2], []string{}
+		for _, ipGroup := range ipMatch.FindAllStringSubmatch(groups[3], -1) {
+			ips = append(ips, ipGroup[1])
+		}
+		result = append(result, psEntry{containerIDPrefix, macAddress, ips})
+	}
+	return result, scanner.Err()
+}
+
+func (w Weave) tagContainer(r report.Report, containerIDPrefix, macAddress string, ips []string) {
+	for nodeid, nmd := range r.Container.NodeMetadatas {
+		idPrefix := nmd.Metadata[docker.ContainerID][:12]
+		if idPrefix != containerIDPrefix {
+			continue
+		}
+
+		existingIPs := report.MakeIDList(strings.Fields(nmd.Metadata[docker.ContainerIPs])...)
+		existingIPs = existingIPs.Add(ips...)
```

Comment:
This is only done in one place (here), so its not clear breaking this out into a function adds anything.